torchdynamo/utils.py

```diff
@@ -74,8 +74,9 @@ def __setitem__(self, key, value):
         self.values[idx] = value
 
     def _remove_id(self, idx):
-        if idx in self.refs:
+        if idx in self.values:
             del self.values[idx]
+        if idx in self.refs:
             del self.refs[idx]
```

Comment:
Hmm, does this feel like its hiding a bug elsewhere in the class, where these are getting out of sync?

Reply:
Yeah, I think it is hiding a bug underneath. I did not get a chance to dig deep here.

Let me convert this PR to WIP, and I will take deeper look.

Reply:
If you throw up a repro case, I can look early next week too!

Reply:
I think this is actually coming from `.clear()` in this same class.  Clear deletes refs then values, when it should do it in opposite order.